lib/publiccloud/provider.pm

```diff
@@ -32,6 +34,17 @@ sub init {
     die('init() isn\'t implemented');
 }
 
+=head2 prepare_os
+
+Required actions to prepare the OS for running the tests
+
+=cut
+sub prepare_os {
```

Comment:
I think `register_os` would be more accurate to what we actually doing here

Reply:
It was a mistake moving it to provider.pm. I have kept it in upload_img, so no modifications in this side for this PR.